server/routes/api/section.js

```diff
@@ -159,4 +165,45 @@ router.get('/updateSection', (req, res) => {
     });
 });
 
-module.exports = router;
\ No newline at end of file
+module.exports = router;
+
+// ------------- Combined Seating Layout and Course Methods -------------
+
+// gets information regarding a course given a student/prof id
+router.get('/getCourseView', (req, res) => {
+    // this is the professor ID
+    let courseID = req.body.courseID;
+
+    Course.findOne( {_id: courseID} , function(err, course){
+        if(err){
+            console.log(err);
+            return res.status(500).send();
+        }
+
+        seatingLayout = SeatingLayout.findOne({_id: course.seating_layout}, function(err, seatingLayout){
+
+            let myLayout;
+            if(err){
+                console.log(err);
+                return res.status(500).send();
+            }
+    
+            if(seatingLayout != null && seatingLayout.layout != null) {
+                myLayout = seatingLayout.layout;
+                console.log("?");
+            } else {
+                myLayout = "Layout not found";
+            }
+
+            return res.status(200).json({
+                "name" : course.name,
+                "alwaysMandatory" : course.always_mandatory,
+                "students": course.students,
```

Comment:
The field students was changed to registered_students in the database.
```suggestion
                "students": course.registered_students,
```